### PR TITLE
Queries + db + web

### PR DESCRIPTION
**Pull request by iamnovik**

### CODE REVIEW

File: online_s/online_s/Controllers/BrandController.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using online_s.ScaffDir;
+namespace online_s.Controllers;
+
```
Comment (Katletos) [inline online_s/online_s/Controllers/BrandController.cs]: set route with annotations

File: online_s/online_s/Startup.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using FluentValidation.AspNetCore;
+using Microsoft.EntityFrameworkCore;
+using online_s.ScaffDir;
+
+namespace online_s
+{
```
Comment (Katletos) [inline online_s/online_s/Startup.cs]: read how it works at .net 7

File: online_s/online_s/Program.cs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        CreateHostBuilder(args).Build().Run();
+
+    }
+}
+
+
+
+
+
+
```
Comment (Katletos) [inline online_s/online_s/Program.cs]: set up linter for code formatting. (delete redurant rows)

File: online_s/online_s/ScaffDir/M2mOrderProduct.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace online_s.ScaffDir;
+
+public partial class M2mOrderProduct
+{
+    public long M2mOrderProductOrderId { get; set; }
```
Comment (Katletos) [inline online_s/online_s/ScaffDir/M2mOrderProduct.cs]: delete redurant entities. You set up m2m relations using fluent api

**Review by Katletos — CHANGES_REQUESTED** [review]

File: OnlineShop/OnlineShop/Models/Configurations/AdressConfiguration.cs [hunk 5]
```diff
@@ -0,0 +1,32 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using online_s.ScaffDir;
```
Comment (Katletos) [inline OnlineShop/OnlineShop/Models/Configurations/AdressConfiguration.cs]: namespace

File: OnlineShop/OnlineShop/Models/Configurations/CategoryConfiguration.cs [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using online_s.ScaffDir;
+
+namespace online_s.Configurations;
+
+public class CategoryConfiguration : IEntityTypeConfiguration<Category>
+{
+    public void Configure(EntityTypeBuilder<Category> entity)
+    {
+        entity.ToTable("Category");
```
Comment (Katletos) [inline OnlineShop/OnlineShop/Models/Configurations/CategoryConfiguration.cs]: delete this shit

File: OnlineShop/OnlineShop/Models/Configurations/SectionConfiguration.cs [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+public class SectionConfiguration : IEntityTypeConfiguration<Section>
+{
+    public void Configure(EntityTypeBuilder<Section> entity)
+    {
+        entity.ToTable("Section");
+
+        entity.Property(e => e.SectionId)
+            .HasDefaultValueSql("nextval(('\"section_section_id_seq\"'::text)::regclass)")
+            .HasColumnName("section_id");
```
Comment (Katletos) [inline OnlineShop/OnlineShop/Models/Configurations/SectionConfiguration.cs]: I don't like hard-coded names. Edit in EACH config file

File: OnlineShop/OnlineShop/Program.cs [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+            .ToList();
+    }
+
+    async Task<IEnumerable<Product>> GetProductBySectionAndCategory(ShopDbContext dbContext, long sectionId,
+        long categoryId)
+    {
+        /*var results = await dbContext.Products
+        .Join(dbContext.M2mSectionCategories, p => p.ProductCategoryId, m => m.M2mSectionCategoryCategoryId, (p, m) => new { Product = p, M2mSectionCategory = m })
+        .Join(dbContext.Sections, pm => pm.M2mSectionCategory.M2mSectionCategorySectionId, s => s.SectionId, (pm, s) => new { pm.Product, pm.M2mSectionCategory, Section = s })
+        .Join(dbContext.Categories, pms => pms.M2mSectionCategory.M2mSectionCategoryCategoryId, c => c.CategoryId, (pms, c) => new { pms.Product, pms.M2mSectionCategory, pms.Section, Category = c })
```
Comment (Katletos) [inline OnlineShop/OnlineShop/Program.cs]: wtf?

File: OnlineShop/OnlineShop/Validators/BrandValidator.cs [hunk 9]
```diff
@@ -0,0 +1,13 @@
+﻿using FluentValidation;
```
Comment (Katletos) [inline OnlineShop/OnlineShop/Validators/BrandValidator.cs:1]: It's very good that you've hooked up validation.

File: OnlineShop/OnlineShop/Models/ScaffDir/ShopDbContext.cs [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+
+    public virtual DbSet<Size> Sizes { get; set; }
+
+    public virtual DbSet<User> Users { get; set; }
+
+    protected override void OnModelCreating(ModelBuilder modelBuilder)
+    {
+        modelBuilder.ApplyConfiguration(new AdressConfiguration());
+
+        modelBuilder.ApplyConfiguration(new BrandConfiguration());
```
Comment (Katletos) [inline OnlineShop/OnlineShop/Models/ScaffDir/ShopDbContext.cs]: You an replace this `modelBuilder.ApplyConfiguration(new ColorConfiguration());` boilerplate using ApplyConfigyrationFromAssembly

File: OnlineShop/OnlineShop/Models/ScaffDir/ShopDbContext.cs [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+
+        modelBuilder.ApplyConfiguration(new SizeConfiguration());
+
+        modelBuilder.ApplyConfiguration(new UserConfiguration());
+        
+        modelBuilder.HasSequence("adress_address_id_seq");
+        modelBuilder.HasSequence("brand_brand_id_seq");
+        modelBuilder.HasSequence("cart_cart_user_id_seq");
+        modelBuilder.HasSequence("category_category_id_seq");
+        modelBuilder.HasSequence("color_color_id_seq");
```
Comment (Katletos) [inline OnlineShop/OnlineShop/Models/ScaffDir/ShopDbContext.cs]: fuck fuck fuck fuck this shit

File: OnlineShop/OnlineShop/Models/ScaffDir/User.cs [hunk 12]
```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace online_s.ScaffDir;
+
+public partial class User
```
Comment (Katletos) [inline OnlineShop/OnlineShop/Models/ScaffDir/User.cs]: why is partial?

File: OnlineShop/OnlineShop/Program.cs [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
+    */
+        return null;
+
+    }
+
+    async Task<IEnumerable<Review>> GetReviewsByProductName(ShopDbContext dbContext, string productName)
+    {
+        var reviews = await dbContext.Products.Include(p => p.Reviews)
+            .FirstOrDefaultAsync(p => p.ProductName == productName);
+        if (reviews != null)
```
Comment (Katletos) [inline OnlineShop/OnlineShop/Program.cs]: can use `is not null`, not `!= null`

File: OnlineShop/OnlineShop/Program.cs [hunk 14]
```diff
@@ -0,0 +1,106 @@
...
+
+        return Enumerable.Empty<Review>();
+    }
+
+    async Task<IEnumerable<Order>> GetOrdersByProductAndStatus(ShopDbContext dbContext, int productId, int orderStatus)
+    {
+
+        var results = await dbContext.Orders
+            .OrderByDescending(o => o.OrderTimeCreate)
+            .Join(dbContext.OrderProducts, o => o.OrderId, m => m.OrderProductOrderId,
```
Comment (Katletos) [inline OnlineShop/OnlineShop/Program.cs]: strange query

File: OnlineShop/OnlineShop/Program.cs [hunk 15]
```diff
@@ -0,0 +1,106 @@
...
+            .Where(result =>
+                result.ProductSizeColor.pvProductId == productId && result.Order.OrderStatus == orderStatus)
+            .Select(result => result.Order)
+            .ToListAsync();
+
+        return results;
+    }
+
+
+    public static void Main(String[] args)
```
Comment (Katletos) [inline OnlineShop/OnlineShop/Program.cs:9]: Holy fuck. Just start new web api project. I want a simple uniform project structure, not a hodgepodge of incomprehensible things

File: OnlineShop/OnlineShop/Validators/BrandValidator.cs [hunk 16]
```diff
@@ -0,0 +1,13 @@
...
+using online_s.ScaffDir;
+
+namespace online_s.Validators;
+
+public class BrandValidator : AbstractValidator<Brand>
+{
+    public BrandValidator()
+    {
+        RuleFor(b => b.BrandName).NotEmpty().MaximumLength(50);
+        RuleFor(b => b.BrandId).NotEmpty();
```
Comment (Katletos) [inline OnlineShop/OnlineShop/Validators/BrandValidator.cs]: add data types for fields

**Review by Katletos — CHANGES_REQUESTED** [review]

File: OnlineShop/OnlineShop/Models/ScaffDir/User.cs (hunk 12 above)
Comment (iamnovik) [reply OnlineShop/OnlineShop/Models/ScaffDir/User.cs]: Classes were partial after scaffolding. I didn't noticed it's become redundant.

File: OnlineShop/OnlineShop/Models/Configurations/AdressConfiguration.cs [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+        entity.HasIndex(e => e.AddressUserId);
+        
+        entity.Property(e => e.AddressId)
+            .UseIdentityColumn()
+            .IsRequired();
+        
+        entity.Property(e => e.AddressAddress)
+            .HasMaxLength(50)
+            .HasColumnName(nameof(Adress.AddressAddress));
+        entity.Property(e => e.AddressUserId).HasColumnName(nameof(Adress.AddressUserId));
```
Comment (Katletos) [inline OnlineShop/OnlineShop/Models/Configurations/AdressConfiguration.cs]: delete `HasColumnName`. DELETE ALL NAMES

File: OnlineShop/OnlineShop/Models/Configurations/BrandConfiguration.cs [hunk 18]
```diff
@@ -0,0 +1,17 @@
...
+namespace online_s.Configurations;
+
+public class BrandConfiguration : IEntityTypeConfiguration<Brand>
+{
+    public void Configure(EntityTypeBuilder<Brand> entity)
+    {
+        entity.Property(e => e.BrandId)
+            .UseIdentityColumn()
+            .IsRequired();
+        entity.Property(e => e.BrandName)
```
Comment (Katletos) [inline OnlineShop/OnlineShop/Models/Configurations/BrandConfiguration.cs:14]: add unique foro name. Check this for other entities

File: OnlineShop/OnlineShop/Models/Configurations/CategoryConfiguration.cs [hunk 19]
```diff
@@ -0,0 +1,28 @@
...
+        entity.Property(e => e.CategoryName)
+            .HasMaxLength(50);
+        entity.Property(e => e.CategorySubcategoryId);
+
+        entity.HasOne(d => d.CategorySubcategory).WithMany(p => p.InverseCategorySubcategory)
+            .HasForeignKey(d => d.CategorySubcategoryId);
+        
+        entity.HasMany(c => c.Sections)
+            .WithMany(s => s.Categories)
+            .UsingEntity(j => j.ToTable("M2M_Sections_Categories"));
```
Comment (Katletos) [inline OnlineShop/OnlineShop/Models/Configurations/CategoryConfiguration.cs]: rewrite `            .UsingEntity(j => j.ToTable("M2M_Sections_Categories"));`

File: OnlineShop/OnlineShop/Models/Configurations/ProductVariantConfiguration.cs [hunk 20]
```diff
@@ -0,0 +1,40 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using OnlineShop.Models.ScaffDir;
+
+namespace online_s.Configurations;
```
Comment (Katletos) [inline OnlineShop/OnlineShop/Models/Configurations/ProductVariantConfiguration.cs]: namespace...

**Review by Katletos — CHANGES_REQUESTED** [review]

File: OnlineShop/OnlineShop/Models/Configurations/OrderProductConfiguration.cs [hunk 21]
```diff
@@ -0,0 +1,30 @@
...
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using OnlineShop.Models.ScaffDir;
+
+namespace OnlineShop.Models.Configurations;
+
+public class OrderProductConfiguration : IEntityTypeConfiguration<OrderProduct>
+{
+    public void Configure(EntityTypeBuilder<OrderProduct> entity)
+    {
+        entity.HasIndex(e => e.OrderProductProductId);
```
Comment (Katletos) [inline OnlineShop/OnlineShop/Models/Configurations/OrderProductConfiguration.cs:11]: it's  also keys

**Review by Katletos — APPROVED** [review]
I like it. You should only reread and fix configurations

**Pull request merged**